tests/test_runner/tests.py

```diff
@@ -407,3 +408,61 @@ def test_empty_default_database(self):
         connection = testcases.connections[db.utils.DEFAULT_DB_ALIAS]
         self.assertEqual(connection.settings_dict['ENGINE'], 'django.db.backends.dummy')
         connections_support_transactions()
+
+
+class Ticket29866RegressionTests(unittest.TestCase):
+    def test_run_checks_raises(self):
+        """
+        Test that when run_checks raises SystemCheckError the teardown functions are run.
```

Comment:
Avoid using `Test that` prefixes in docstring, this is necessarily testing something as it is a test.

```
Teardown functions are run when run_checks raises SystemCheckError.
```